cheetah/accelerator.py

```diff
@@ -58,13 +63,41 @@ def __init__(self, name=None, device="auto"):
         self.device = device
 
     def transfer_map(self, energy):
+        """
+        Generates the element's transfer map that describes how the beam and its particles are transformed when travelling through the element.
+        The Matrix R that is returned is then multiplied with the state vector in the function '__call__'.
+        The state vector is consisting of 6 values with a physical meaning:
+            x: Position in x direction
+            xp: Momentum in x direction
+            y: Position in y direction
+            yp: Momentum in y direction
+            s: Position in z direction, the zero value is set to the middle of the pulse
+            sp: Momentum in s direction
+        As well as a seventh value used to add constants to some of the prior values if necesassary. Through this
+        seventh state, the addition of constants can be represented using a matrix multiplication.
+        
+        Parameters
+        ----------
+        energy: cheetah.Beam.energy
+            Energy of the Beam. Read from the fed in Cheetah Beam.
+           
+        Returns
+        --------
+        R  
+            A 7x7 Matrix for further calculations.
+            
+        Raises
+        -------
+        NotImplementedError
+            If no transfer_map function is implemented for the given `Element` subclass.
+        """
         raise NotImplementedError
 
     def __call__(self, incoming):
         """
-        Track particles through the element.
+        Track particles through the element. Parameter and Particle Beams are treated differently, using an if statement. 
```

Comment:
The user does not need to know how the method works, i.e. that there is an `if` in it. Maybe just mention that it can handle `ParticleBeam` and `ParameterBeam`. It's also good practice to call them by there class names in reverse ticks.